internal/dag/httpproxy_processor.go

```diff
@@ -640,6 +653,11 @@ func (p *HTTPProxyProcessor) computeRoutes(
 				r.Clusters = append(r.Clusters, c)
 			}
 		}
+		if len(r.Clusters) == 0 {
```

Comment:
We should baby step this, but the next one would be to handle multiple services in the same route and setting the invalid one to a weight of zero.

Reply:
I wonder, will the invalid service be programmed at all, i.e. does it exist from Envoy perspective?

The simplest case seemed to work already now: I had two services for a route, one of those missing (and of course missing service cannot be programmed to Envoy).

Reply:
Yes, there wouldn't be a cluster programming in envoy matching the missing service, but I believe our route logic would still try and program it. We could change that logic to not program a the route with a missing service or set the weight to zero. 

Either way we'll need a way to understand that the service is invalid somehow when building out the route. But this can be a different PR. This one handling the case for a single service, then another PR to cover multiple services cases.

Reply:
I did not manage to find where in the route logic missing services should be deal. Though I have some difficulties understanding the difference between cluster weight (`Cluster.Weight`) vs service weight `Cluster.Upstream.Weighted.Weight`, so I might be missing some key piece here...

Meanwhile, I added test case "root proxy refers to three services with weights, one is missing", where the missing service just results in one less cluster.  The weights of existing clusters are kept like they were in the spec.

Reply:
> Though I have some difficulties understanding the difference between cluster weight (Cluster.Weight) vs service weight Cluster.Upstream.Weighted.Weight, so I might be missing some key piece here...

No I agree that's confusing and never got cleaned up: https://github.com/projectcontour/contour/issues/3593